src/overlays/actors/ovl_En_Go/z_en_go.c

```diff
@@ -52,93 +76,1173 @@ static ColliderCylinderInit D_80A41B00 = {
+                            if (CUR_UPG_VALUE(UPG_STRENGTH) || (gSaveContext.infTable[14] & 0x800)) {
+                                this->actor.textId = 0x300B;
+                            } else {
+                                this->actor.textId = 0x300C;
+                            }
+                        } else {
+                            this->actor.textId = 0x300D;
+                        }
+                        func_8010B720(globalCtx, this->actor.textId);
+                        unkState = 1;
+                        break;
+                    case 0x3034:
+                        if (globalCtx->msgCtx.choiceIndex == 0) {
+                            if (gSaveContext.infTable[16] & 0x800) {
+                                this->actor.textId = 0x3033;
+                            } else {
+                                this->actor.textId = 0x3035;
+                            }
+                        } else if (gSaveContext.infTable[16] & 0x800) {
+                            this->actor.textId = 0x3036;
+                        } else {
+                            this->actor.textId = 0x3033;
+                        }
+                        func_8010B720(globalCtx, this->actor.textId);
+                        unkState = 1;
+                        break;
+                    case 0x3054:
+                    case 0x3055:
+                        if (globalCtx->msgCtx.choiceIndex == 0) {
+                            unkState = 2;
+                        } else {
+                            this->actor.textId = 0x3056;
+                            func_8010B720(globalCtx, this->actor.textId);
+                            unkState = 1;
+                        }
+                        gSaveContext.infTable[11] |= 0x10;
+                        break;
+                }
+            }
+            break;
+        case 5:
+            if (func_80106BC8(globalCtx)) {
+                switch (this->actor.textId) {
+                    case 0x3035:
+                        gSaveContext.infTable[16] |= 0x800;
+                    case 0x3032:
+                    case 0x3033:
+                        this->actor.textId = 0x3034;
+                        func_8010B720(globalCtx, this->actor.textId);
+                        unkState = 1;
+                        break;
+                    default:
+                        unkState = 2;
+                        break;
+                }
+            }
+            break;
+        case 6:
+            if (func_80106BC8(globalCtx)) {
+                unkState = 3;
+            }
+            break;
+        case 0:
+        case 1:
+        case 3:
+        case 7:
+        case 9:
+            break;
+    }
+    return unkState;
+}
+
+s32 func_80A3ED24(GlobalContext* globalCtx, EnGo* this, struct_80034A14_arg1* arg2, f32 arg3,
+                  u16 (*GetTextID)(GlobalContext*, Actor*), s16 (*unkFunc2)(GlobalContext*, Actor*)) {
```

Comment:
These arguments should probably use function pointer types, but either way the name should fit our scheme
```suggestion
                  u16 (*getTextId)(GlobalContext*, Actor*), s16 (*unkFunc2)(GlobalContext*, Actor*)) {
```